packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart

```diff
@@ -425,33 +428,42 @@ class DynamiteClient {
     return executeRawRequest(
       method,
       uri,
-      {...?baseHeaders, ...headers},
+      {
+        if (userAgent != null) 'user-agent': userAgent!,
+        ...?baseHeaders,
+        ...headers,
+      },
       body,
       validStatuses,
     );
   }
 
   /// Executes a HTTP request against give full [uri].
-  Future<HttpClientResponse> executeRawRequest(
+  Future<http.StreamedResponse> executeRawRequest(
     String method,
     Uri uri,
     Map<String, String> headers,
     Uint8List? body,
     Set<int>? validStatuses,
   ) async {
-    final request = await httpClient.openUrl(method, uri);
+    final request = http.Request(method, uri);
     request.headers.addAll(headers);
 
     if (body != null) {
-      request.add(body);
+      request.bodyBytes = body;
     }
+
     if (cookieJar != null) {
-      request.cookies.addAll(await cookieJar!.loadForRequest(uri));
+      final cookies = await cookieJar!.loadForRequest(uri);
+      request.headers['cookie'] = cookies.join('; ');
     }
 
-    final response = await request.close();
-    if (cookieJar != null) {
-      await cookieJar!.saveFromResponse(uri, response.cookies);
+    final response = await httpClient.send(request);
+
+    final cookieHeader = response.headers['set-cookie'];
+    if (cookieHeader != null && cookieJar != null) {
+      final cookies = cookieHeader.split(RegExp(',(?=[^ ])')).map(Cookie.fromSetCookieValue).toList();
```

Comment:
This looks dangerous, does it follow some standard on how to split cookies or did you copy this implementation from somewhere else where it is known to work?

Reply:
Neat :D